src/smt/preprocess_proof_generator.cpp

```diff
@@ -63,7 +65,10 @@ void PreprocessProofGenerator::notifyNewAssert(Node n, ProofGenerator* pg)
     // if no proof generator provided for (non-true) assertion
     if (pg == nullptr)
     {
-      checkEagerPedantic(d_ra);
+      Assert(id != TrustId::UNKNOWN_PREPROCESS_LEMMA);
```

Comment:
isn't this gonna be the solver kind of unusable if one wants a production + assertion build? You could guard this instead with whether the build is a DEBUG build, not only via an assertion. Or even with an option, like with what we had before with pedantic levels...

Reply:
I don't understand the suggestion here.  This is a normal assertion?

Reply:
This is a debug assertion failure, it only applies to debug builds.

Moreover this is the intention moving forward, that we want debug assertions to throw in any case where the lack of proof production could (and should) be considered an error.  Nevertheless this assertion does not throw in production currently.

Reply:
But this does lead to a crash if assertions are on currently, yes? My point is that if you want to add an assertion, is to something that we don't want to happen that do not currently happen.

Reply:
Assertions are checked by CI and this does not break CI :)
The only way this should break is if someone added new code (e.g. a new preprocessing pass) that did not introduce the proper trust id when being used with proofs.

I've added more trust ids in my last commit to handle even the preprocessing passes that cannot be combined with proofs.

Reply:
I see, I misunderstood that those preprocessing passes could lead to these issues.